### PR TITLE
BATCH-2380: disallow null + empty delimiter

### PR DESCRIPTION
**Pull request by mouyang**
I have signed and agree to the terms of the SpringSource Individual Contributor License Agreement.

### CODE REVIEW

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/DelimitedLineTokenizer.java [hunk 1]
```diff
@@ -133,6 +134,8 @@ public void setQuoteCharacter(char quoteCharacter) {
 	@Override
 	protected List<String> doTokenize(String line) {
 
+		Assert.state(null != delimiter && 0 != delimiter.length());
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/DelimitedLineTokenizer.java]: We don't want to be doing this for every record.  We also would want to know there's an issue before this point.  What about implementing `InitializerBean` and providing the check there?

Comment (mouyang) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/DelimitedLineTokenizer.java]: I agree.  It is not expected that the delimiter will change after a bean has been initialized.

**Comment by mminella** [comment]
Two last, minor, requests:  

1. Can you create this PR against master instead of 3.0.x?  While the null check isn't really a breaking change, the empty string technically is.
2.  Going forward, the convention for branching is to name the branch after the Jira issue it's solving.  So in your case, the branch would be called BATCH-2380.

**Comment by mouyang** [comment]
will do

**Pull request closed without merging**